### PR TITLE
Differential Team into master

### PR DESCRIPTION
**Pull request by rlee287**

### CODE REVIEW

**Comment by rlee287** [comment]
This can be merged either before or after the hotfix PR #208.

File: goon-universe/queries.js [hunk 1]
```diff
@@ -70,7 +76,17 @@ function getTaskPage(scenarioID, callback){
             throw error
         }
         callback(results.rows)
-    }) 
+    })
+}
+
+function getConclusionPage(scenarioID, callback){
```
Comment (rasdy) [inline goon-universe/queries.js:84]: Calling `GET /scenarios/conclusion` with `scenarioID: 1` returns a 404. Is the example data set up correctly to test this endpoint?

File: goon-universe/queries.js [hunk 2]
```diff
@@ -241,35 +257,78 @@ function addFinalReflectResponse(studentID, input, promptNum, scenarioID, timest
...
 }
 
+
+// scenarioExists(scenarioID)
+// .then(function(result){
+//     console.log(result);
+// })
+// .catch(function(err){
+//     console.log(err);
+// });
```
Comment (rasdy) [inline goon-universe/queries.js:269]: Are the commented-out functions like this left in on purpose?

File: goon-universe/queries.js [hunk 3]
```diff
@@ -614,45 +689,25 @@ function getFinalReflectPage(scenarioID, callback){
...
+function getInitActionPageQuestionsAndChoices(scenarioID,questionID,callback){
+    let thisQuery='select question.question, question.id, mcq_option.option from pages, mcq, mcq_option, question where pages.order ='+ INIT_ACTION +'and scenario_id = $1 and mcq.page_id = question.mcq_id and pages.id=mcq.page_id and mcq_option.question_id = $2 and mcq_option.question_id = question.id'
+    pool.query(thisQuery, [scenarioID, questionID], (error,results) => {
+        if(error){
             throw error
         }
         callback(results.rows)
-    })  
+    })
 }
```
Comment (rasdy) [inline goon-universe/queries.js]: Why is `getInitActionPageQuestionsAndChoices()` taking in a questionID parameter, but also returing question IDs?

**Review by rasdy — CHANGES_REQUESTED** [review]
Had a couple questions and brought up an issue. Someone from track 4 should review this as well, since I wouldn't know what to look for regarding the editor-related functions.

File: goon-universe/queries.js (hunk 2 above)
Comment (rlee287) [reply goon-universe/queries.js:269]: This comment explains how to use the scenarioExists function that is defined below.

File: goon-universe/queries.js (hunk 1 above)
Comment (rlee287) [reply goon-universe/queries.js:84]: Probably; I can double-check this once I update the example data

File: goon-universe/queries.js (hunk 3 above)
Comment (rlee287) [reply goon-universe/queries.js]: Fixed in 05936594.

File: goon-universe/queries.js (hunk 1 above)
Comment (rlee287) [reply goon-universe/queries.js:84]: Test data is not fully updated but this particular endpoint should work correctly now. Once you test this again we can resolve this conversation.

**Comment by rlee287** [comment]
Pinging @aj8uppal for code review later (not yet as we are still fixing some things)

File: goon-universe/queries.js (hunk 1 above)
Comment (rlee287) [reply goon-universe/queries.js:84]: @rasdy can you reverify when you have a chance?

Comment (rasdy) [reply goon-universe/queries.js:84]: Works fine. Ping me on slack next time I'm slow to respond on github

**Review by aj8uppal — DISMISSED** [review]
Looks good to me.

File: goon-universe/app.js [hunk 4]
```diff
@@ -225,7 +225,7 @@ router.route('/scenarios/initialAction')
             res.end()
         }
         else{
-        db.getInitActions(scenarioID, function(result){
+        db.getInitActionPageQuestionsAndChoices(scenarioID, function(result){
```
Comment (rasdy) [inline goon-universe/app.js:228]: Querying `GET /scenarios/initialAction` with `scenarioID: 1, studentID: 1` results in an unhandled exception being thrown.

File: goon-universe/app.js [hunk 5]
```diff
@@ -252,7 +252,7 @@ router.route('/scenarios/initialAction')
             res.end()
         }
         else{
-        db.addInitActionChoice(studentID, scenarioID, data, function(result){
+        db.addInitActionResponse(studentID, scenarioID, data, function(result){
```
Comment (rasdy) [inline goon-universe/app.js]: `addInitActionResponse()` takes in more arguments than this.

**Review by rasdy — CHANGES_REQUESTED** [review]
Found a couple issues

File: goon-universe/app.js (hunk 4 above)
Comment (kungoyal) [reply goon-universe/app.js:228]: DT's implementation of the getInitActionPageQuestionsAndChoices takes questionID as an argument along with scenarioID. I believe this is the issue. Also this function does not require studentID. @rasdy @rlee287

File: goon-universe/app.js (hunk 5 above)
Comment (kungoyal) [reply goon-universe/app.js]: Fixed

Comment (kungoyal) [reply goon-universe/app.js]: This has been implemented similar to addXReflectResponse functions with data being an object with questionID as key and choiceID as value

File: goon-universe/app.js (hunk 4 above)
Comment (rlee287) [reply goon-universe/app.js:228]: Function now does callback on empty object (if question and options do not exist) or on JSON object with `question` and `mcq_choices` fields. I will ping both of you on Slack once I finish these changes and push them.

Comment (rlee287) [reply goon-universe/app.js:228]: This should be fixed now.

File: goon-universe/app.js [hunk 6]
```diff
@@ -226,7 +226,7 @@ router.route('/scenarios/initialAction')
         }
         else{
         db.getInitActionPageQuestionsAndChoices(scenarioID, function(result){
```
Comment (rasdy) [inline goon-universe/app.js:228]: The action GET endpoints should return the question and choice IDs that the action PUT endpoint expects.

**Review by rasdy — CHANGES_REQUESTED** [review]

File: goon-universe/app.js (hunk 6 above)
Comment (rlee287) [reply goon-universe/app.js:228]: This should be fixed by the latest commit.

**Review by rasdy — APPROVED** [review]
Looks good

**Pull request merged**